### PR TITLE
Issue5

### LINKED ISSUE #5: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by aman-singh**
Closes #5
Filtering images by tag

### CODE REVIEW

**Comment by aman-singh** [comment]
[Issue 5: Images by Tag](https://trello.com/c/hA5w1Jxw/793-issue-5-images-by-tag)

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<h1>Aman's Image Sharer</h1>
+<h4>Images with tag: <%= @tag %></h4>
+<br/>
+<% @images.each do |image| %>
+  <p>Create Date: <%= image.created_at %>></p>
+  <%= image_tag image.url, class: 'index-img' %>
+  <p>Tags: <% image.tag_list.each do |tag| %>
+      <%= link_to tag, controller: :images, tag: tag %>
```
Comment (sthust) [inline app/views/images/index.html.erb:8]: pls add an id to the link_to element produced, e.g. class: "js-image_tag"

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -36,4 +36,20 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    end
+  end
+
+  test 'should display images for a given tag' do
+    Image.create! url: 'https://cdn.pixabay.com/736885_960_720.jpg', tag_list: 'tag1'
+    Image.create! url: 'https://cdn.pixabay.com/736885_960_720.jpg', tag_list: 'tag2'
+    get images_path, params: { tag: :tag1 }
+    assert_response :success
+    assert_select 'img', 1 do |elements|
+      assert_equal 'https://cdn.pixabay.com/736885_960_720.jpg', elements[0].attributes['src'].value
```
Comment (sthust) [inline test/controllers/images_controller_test.rb:52]: you can use the previously added "js-image_tag" if to more reliably detect the element.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -36,4 +36,20 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+  test 'should fail if tag query param missing for index api call' do
+    assert_raises(ActionController::ParameterMissing) do
+      get images_path
+    end
+  end
+
+  test 'should display images for a given tag' do
+    Image.create! url: 'https://cdn.pixabay.com/736885_960_720.jpg', tag_list: 'tag1'
+    Image.create! url: 'https://cdn.pixabay.com/736885_960_720.jpg', tag_list: 'tag2'
+    get images_path, params: { tag: :tag1 }
```
Comment (sthust) [inline test/controllers/images_controller_test.rb:49]: might be good to add a few additional test cases:
1) another image with the same tag, e.g. 'tag1' to test that both images are returned. 
2) another image with different multiple tags e.g.  tag 3, tag 4 and assert that it's not showing up if you look e.g. for tag 1
3) no image returned for a tag 
you could even add an extra notice message if no image was found for a tag, but that's not required.

File: app/views/images/index.html.erb (hunk 1 above)
Comment (aman-singh) [reply app/views/images/index.html.erb]: done

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (aman-singh) [reply test/controllers/images_controller_test.rb]: done

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (aman-singh) [reply test/controllers/images_controller_test.rb:51]: Thanks! I think I covered all the cases mentioned here in the latest update, let me know if that's not the case.

**Pull request merged**